example/snippets.cpp

```diff
@@ -0,0 +1,243 @@
+//
+// Copyright (c) 2022 Alan Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/CppAlliance/url
+//
+
+//[snippet_headers_1
+#include <boost/url.hpp>
+//]
+
+#if 0
+//[snippet_headers_2
+#include <boost/url/src.hpp>
+//]
+#endif
+
+#include <iostream>
+
+//[snippet_headers_3
+#include <boost/url.hpp>
+using namespace boost::urls;
+//]
+
+void
+using_url_views() {
+    //[snippet_parsing_1
+    string_view url_str = "https://user:pass@www.example.com:443/path/to/my%2dfile.txt?id=42&name=John%20Doe#page%20anchor";
```

Comment:
I prefer `s` over `url_str`, because `url_str` is far too busy, and it looks like a type from the library. How do we know it isn't `urls::url_str`? It would make sense if it was.